app/services/moderator/banisher.rb

```diff
@@ -40,6 +44,8 @@ def add_banned_role
     end
 
     def delete_reactions
+      return unless user.reactions.count.positive?
+
       user.reactions.find_each(&:delete)
```

Comment:
Why not `delete_all` , if it's `delete`. not `destroy` anyway?
With `delete_all` the first check won't be needed probably.

Reply:
I went with `delete_all` and removed the checks for most. For the ones I kept checks for, I optimized the query per your suggestion. Thank you Anna!